test/unity/toolkits/object_detection/one_shot_object_detection/test_parameter_sampler.cxx

```diff
@@ -0,0 +1,105 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#define BOOST_TEST_MODULE test_parameter_sampler
+
+#include <toolkits/object_detection/one_shot_object_detection/util/parameter_sampler.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include <core/util/test_macros.hpp>
+
+#include <unordered_map>
+
+static constexpr int max_dimension = 6000;
+static constexpr int num_rows = 1000;
+
+namespace turi {
+namespace one_shot_object_detection {
+namespace {
+
+ParameterSampler create_parameter_sampler(std::default_random_engine &engine) {
+
+  std::uniform_int_distribution<int> dist(0, max_dimension);
+  
+  size_t starter_width = dist(engine);
+  size_t starter_height = dist(engine);
+  size_t dx = 0;
+  size_t dy = 0;
+  ParameterSampler sampler(starter_width, starter_height, dx, dy);
+  return sampler;
+}
+
+std::unordered_map<std::string, std::vector<double> > get_all_angles(
```

Comment:
`get_all_angles` seems like a strange name for this function. `get` normally connotes a simple accessor. Here you're actually sampling from the `sampler`

Reply:
Changed it to `construct_angle_samples_map`.